pkg/condition/eval/eval.go

```diff
@@ -16,11 +17,19 @@ func TupleConditionMet(
 	tupleCondition := tupleKey.GetCondition()
 	conditionName := tupleCondition.GetName()
 	if conditionName != "" {
-		evaluableCondition := typesys.GetCondition(conditionName)
-		conditionResult, err := evaluableCondition.Evaluate(
-			context,
-			tupleCondition.GetContext().AsMap(),
-		)
+		evaluableCondition, ok := typesys.GetCondition(conditionName)
+		if !ok {
+			return false, fmt.Errorf("failed to evaluate relationship condition: condition '%s' was not found", conditionName)
+		}
+
+		// merge both contexts
+		contextSlice := []map[string]interface{}{context}
+		tupleContext := tupleCondition.GetContext()
+		if tupleContext != nil {
+			contextSlice = append(contextSlice, tupleContext.AsMap())
+		}
+
+		conditionResult, err := evaluableCondition.Evaluate(contextSlice...)
```

Comment:
fwiw `.Evaluate()` already merges the passed in context maps

Reply:
That's ok, the change here is to prevent panic when `tupleCondition.GetContext()` is `nil`, since calling `.AsMap(` on it will panic